manager/allocator/cnmallocator/networkallocator.go

```diff
@@ -930,10 +930,29 @@ func initializeDrivers(reg *drvregistry.DrvRegistry) error {
 
 func serviceNetworks(s *api.Service) []*api.NetworkAttachmentConfig {
 	// Always prefer NetworkAttachmentConfig in the TaskSpec
+	networks := s.Spec.Task.Networks
+	filteredNetworks := []*api.NetworkAttachmentConfig{}
 	if len(s.Spec.Task.Networks) == 0 && len(s.Spec.Networks) != 0 {
-		return s.Spec.Networks
+		networks = s.Spec.Networks
 	}
-	return s.Spec.Task.Networks
+	for _, n := range networks {
+		if IsPredefinedNetwork(n.Target) {
+			continue
+		}
+		filteredNetworks = append(filteredNetworks, n)
+	}
+	return filteredNetworks
+}
+
+func taskNetworks(t *api.Task) []*api.NetworkAttachment {
```

Comment:
I'd prefer to just add:

```go
		if IsPredefinedNetwork(n.Network.ID) {
			continue
		}
```

to loops over `t.Networks` which should skip predefined networks, including `releaseEndpoints`. That seems a little more explicit to me.

Otherwise, we should rename this so it's clear that it's only returning non-predefined networks.